### PR TITLE
feat(core): Add revalidation routes for store status and routes

### PR DESCRIPTION
**Pull request by christensenep**
## What/Why?
Adds api routes to revalidate the route and store status caches in the KV store. The middlewares now behave in an SWR fashion (once hydrated), firing off a revalidation request when fetching a stale cache value (determined by a new `expiryTime` field in the cached value), but using that cached value for the current request. We also no longer await when setting the cache values in the middleware when there is a cache miss.

I've also consolidated the middleware, such that the maintenance mode check now happens in `with-custom-urls`. This allows us to do a single `mget` request to the KV store, which is very significant as the vercel KV store can take >100ms for a single request.

NOTE: Currently the dev adapter is not handling this well, as the KV store in memory for the middleware is a different object than the one the api routes have. Not sure what the solution is here. It doesn't actually break, but effectively the cache persists forever.

## Testing
Tested locally, both with the dev adapter and the vercel development adapter, as well as in a vercel deploy

### CODE REVIEW

File: apps/core/lib/kv/adapters/vercel.ts [hunk 1]
```diff
@@ -9,12 +9,12 @@ export class VercelKvAdapter implements KvAdapter {
...
-    key: string,
-    value: Data,
-    opts?: Options,
-  ) {
-    const response = await this.adapter.set(key, value, opts);
+  async mget<Data>(...keys: string[]) {
+    return this.adapter.mget<Data[]>(keys);
+  }
+
+  async set<Data>(key: string, value: Data) {
```
[the remark this answers is not in the document]
Comment (deini) [reply apps/core/lib/kv/adapters/vercel.ts:16]: Yep I agree with the SWR-like approach, however, this is an adapter and IMO should work as regular KV. When you want swr-like we can not pass `ex`.

File: apps/core/lib/kv/adapters/dev.ts [hunk 2]
```diff
@@ -26,19 +25,19 @@ export class DevKvAdapter implements KvAdapter {
...
 
-    if (entry.expiresAt < Date.now()) {
-      return null;
-    }
-
     // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
     return entry.value as Data;
   }
 
-  async set<Data>(key: string, value: Data, options: { ex?: number } = {}) {
```
[the remark this answers is not in the document]
Comment (deini) [reply apps/core/lib/kv/adapters/dev.ts:37]: My bad `expiryTime` is not related to kv, just something internal we store. My comment of removing `ex` remains tho.

File: apps/core/lib/kv/adapters/vercel.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (christensenep) [reply apps/core/lib/kv/adapters/vercel.ts:16]: Gotcha, fair enough

**Comment by christensenep** [comment]
♻️

**Review by deini — APPROVED** [review]

**Review by deini — APPROVED** [review]

**Pull request merged**